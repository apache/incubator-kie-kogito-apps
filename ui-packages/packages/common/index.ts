export {
  default as DataTable
} from './src/components/Organisms/DataTable/DataTable';
export {
  default as KogitoSpinner
} from './src/components/Atoms/KogitoSpinner/KogitoSpinner';
export {
  default as PageToolbar
} from './src/components/Molecules/PageToolbar/PageToolbar';
export {
  default as KogitoPageLayout
} from './src/components/Templates/KogitoPageLayout/KogitoPageLayout';
export {
  default as AboutModalBox
} from './src/components/Molecules/AboutModalBox/AboutModalBox';
export {
  default as EndpointLink
} from './src/components/Atoms/EndpointLink/EndpointLink';
export {
  default as PageNotFound
} from './src/components/Molecules/PageNotFound/PageNotFound';
export {
  default as ServerUnavailable
} from './src/components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './src/components/Molecules/NoData/NoData';
export {
  default as ServerErrors
} from './src/components/Molecules/ServerErrors/ServerErrors';
export {
  default as ProcessDescriptor
} from './src/components/Molecules/ProcessDescriptor/ProcessDescriptor';
export * from './src/components/Atoms/KogitoEmptyState/KogitoEmptyState';
export { default as LoadMore } from './src/components/Atoms/LoadMore/LoadMore';
export {
  default as DomainExplorer
} from './src/components/Organisms/DomainExplorer/DomainExplorer';
export {
  default as DomainExplorerListDomains
} from './src/components/Organisms/DomainExplorerListDomains/DomainExplorerListDomains';
export * from './src/utils/OuiaUtils';
export * from './src/graphql/types';
