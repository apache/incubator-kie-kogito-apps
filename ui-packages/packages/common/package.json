{
  "name": "@kogito-apps/common",
  "version": "1.0.0",
  "private": true,
  "main": "index.js",
  "author": "",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/kiegroup/kogito-apps.git"
  },
  "scripts": {
    "precommit": "lint-staged",
    "build:prod": "yarn run lint && yarn run test:coverage",
    "test:report": "yarn test --ci --reporters=jest-junit",
    "test": "jest --runInBand --ci --reporters=default --reporters=jest-junit",
    "update-snapshot": "jest --updateSnapshot",
    "test:coverage": "yarn test --coverage",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "lint": "tslint -c ./tslint.json --project . './src/**/*.ts{,x}'",
    "format": "prettier --config ../../.prettierrc --check --write './src/**/*.{tsx,ts,js}'",
    "clean": "rimraf dist",
    "codegen": "graphql-codegen"
  },
  "dependencies": {},
  "devDependencies": {
    "@storybook/addon-a11y": "5.3.19",
    "@storybook/addon-actions": "5.3.18",
    "@storybook/addon-backgrounds": "5.3.19",
    "@storybook/addon-console": "1.2.1",
    "@storybook/addon-knobs": "5.3.18",
    "@storybook/addon-links": "5.3.18",
    "@storybook/addon-options": "5.3.18",
    "@storybook/addon-storysource": "5.3.18",
    "@storybook/addon-viewport": "5.3.18",
    "@storybook/react": "5.3.18",
    "@storybook/source-loader": "5.3.18",
    "babel-loader": "8.1.0",
    "react-docgen-typescript-loader": "3.7.2",
    "storybook-react-router": "1.0.8"
  },
  "jest": {
    "preset": "ts-jest/presets/js-with-ts",
    "setupFiles": [
      "../../config/Jest-config/test-shim.js",
      "../../config/Jest-config/test-setup.js",
      "core-js"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "coveragePathIgnorePatterns": [
      "./src/static",
      "./src/graphql"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest",
      "\\.(jpg|jpeg|png|svg)$": "../../config/Jest-config/fileMocks.js"
    },
    "testMatch": [
      "**/tests/*.(ts|tsx|js)"
    ],
    "moduleNameMapper": {
      "\\.(scss|sass|css)$": "identity-obj-proxy"
    }
  },
  "lint-staged": {
    "*.{ts,tsx}": [
      "yarn run format",
      "yarn run lint"
    ]
  }
}
