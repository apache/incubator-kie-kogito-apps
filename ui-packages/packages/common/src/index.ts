/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export {
  default as DataTable,
  DataTableColumn
} from './components/Organisms/DataTable/DataTable';
export { default as KogitoSpinner } from './components/Atoms/KogitoSpinner/KogitoSpinner';
export { default as PageToolbar } from './components/Molecules/PageToolbar/PageToolbar';
export { default as KogitoPageLayout } from './components/Templates/KogitoPageLayout/KogitoPageLayout';
export { default as AboutModalBox } from './components/Molecules/AboutModalBox/AboutModalBox';
export { default as EndpointLink } from './components/Atoms/EndpointLink/EndpointLink';
export { default as PageNotFound } from './components/Molecules/PageNotFound/PageNotFound';
export { default as ServerUnavailable } from './components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './components/Molecules/NoData/NoData';
export { default as ServerErrors } from './components/Molecules/ServerErrors/ServerErrors';
export { default as ItemDescriptor } from './components/Molecules/ItemDescriptor/ItemDescriptor';
export * from './components/Atoms/KogitoEmptyState/KogitoEmptyState';
export { default as LoadMore } from './components/Atoms/LoadMore/LoadMore';
export { default as DomainExplorer } from './components/Organisms/DomainExplorer/DomainExplorer';
export { default as DomainExplorerListDomains } from './components/Organisms/DomainExplorerListDomains/DomainExplorerListDomains';
export * from './utils/Utils';
export * from './environment/auth/Auth';
export {
  default as KogitoAppContext,
  AppContext,
  useKogitoAppContext
} from './environment/context/KogitoAppContext';
export * from './utils/KeycloakClient';
export * from './graphql/types';
export { default as KogitoAppContextProvider } from './components/Molecules/KogitoAppContextProvider/KogitoAppContextProvider';
