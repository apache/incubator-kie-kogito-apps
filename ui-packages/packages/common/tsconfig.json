{
    "extends": "../../tsconfig.json",
    "compilerOptions": {
        "declarationDir": "./dist",
        "declaration": true
    },
    "files": [
        "./index.ts"
    ],
    "include": [
        "src/",
        "../../config/declare.d.ts"
    ],
    "exclude": [
        "src/graphql/queries.tsx",
        "src/**/*.test.ts",
        "src/**/*.test.tsx",
        "src/**/*.stories.tsx",
        "src/**/__mocks__"
    ] 
}
