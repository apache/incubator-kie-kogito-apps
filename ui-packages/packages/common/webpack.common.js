const path = require('path');
const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
const BG_IMAGES_DIRNAME = 'bgimages';

module.exports = {
  entry: {
    index: path.resolve(__dirname, 'index.ts')
  },
  devtool: "inline-source-map",
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: '[name].bundle.js',
    libraryTarget: "umd",
    globalObject: "this"
  },
  module: {
    rules: [
      {
        test: /\.(tsx|ts)?$/,
        use: [
          {
            loader: 'ts-loader',
            options: {
              configFile: path.resolve('./tsconfig.json'),
            }
          }
        ]
      },
      {
        test: /\.(svg|jpg|ttf|eot|woff|woff2)$/,
        include: [
          path.resolve('../../node_modules/patternfly/dist/fonts'),
          path.resolve(
            '../../node_modules/@patternfly/react-core/dist/styles/assets/fonts'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-core/dist/styles/assets/pficon'
          ),
          path.resolve(
            '../../node_modules/@patternfly/patternfly/assets/fonts'
          ),
          path.resolve(
            '../../node_modules/@patternfly/patternfly/assets/pficon'
          ),
          path.resolve('./src/static')
        ],
        use: {
          loader: 'file-loader',
          options: {
            // Limit at 50k. larger files emited into separate files
            limit: 5000,
            outputPath: 'src/static',
            name: '[name].[ext]'
          }
        }
      },
      {
        test: /\.svg$/,
        include: input => input.indexOf('background-filter.svg') > 1,
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 5000,
              outputPath: 'svgs',
              name: '[name].[ext]'
            }
          }
        ]
      },
      {
        test: /\.svg$/,
        include: input => input.indexOf(BG_IMAGES_DIRNAME) > -1,
        use: {
          loader: 'svg-url-loader',
          options: {}
        }
      },
      {
        test: /\.(jpg|jpeg|png|gif)$/i,
        include: [
          path.resolve(__dirname, 'src'),
          path.resolve('../../node_modules/patternfly'),
          path.resolve(
            '../../node_modules/@patternfly/patternfly/assets/images'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-styles/css/assets/images'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-core/dist/styles/assets/images'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-core/node_modules/@patternfly/react-styles/css/assets/images'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-table/node_modules/@patternfly/react-styles/css/assets/images'
          ),
          path.resolve('../../../../../../packages/common/src/static/'),
          path.resolve('./src/static'),
        ],
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 5000,
              outputPath: 'src/static',
              name: '[name].[ext]'
            }
          }
        ]
      }
    ]
  },
  resolve: {
    extensions: ['.ts', '.tsx', '.js'],
    modules: [
      path.resolve('../../node_modules'),
      path.resolve('./node_modules'),
      path.resolve('./src')
    ],
    plugins: [
      new TsconfigPathsPlugin({
        configFile: path.resolve(__dirname, './tsconfig.json')
      })
    ],
    symlinks: false,
    cacheWithContext: false
  }
};