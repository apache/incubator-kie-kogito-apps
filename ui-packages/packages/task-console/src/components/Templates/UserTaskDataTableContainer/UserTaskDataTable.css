.kogito-task-console--user-task_table-OverFlow{
  overflow: auto;
}
