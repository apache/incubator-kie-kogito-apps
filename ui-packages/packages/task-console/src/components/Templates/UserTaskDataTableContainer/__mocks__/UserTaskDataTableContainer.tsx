import React from 'react';

const MockUserTaskDataTableContainer = (): React.ReactElement => {
  return <></>;
};

export default MockUserTaskDataTableContainer;
