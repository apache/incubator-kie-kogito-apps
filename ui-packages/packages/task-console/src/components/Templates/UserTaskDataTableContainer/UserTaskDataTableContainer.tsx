import {
  Card,
  Grid,
  GridItem,
  PageSection,
  InjectedOuiaProps,
  withOuiaContext,
  Bullseye,
  Label
} from '@patternfly/react-core';
import React, { useEffect } from 'react';
import UserTaskPageHeader from '../../Molecules/UserTaskPageHeader/UserTaskPageHeader';
import './UserTaskDataTable.css';
import { useGetUserTasksByStatesQuery } from '../../../graphql/types';
import {
  ouiaPageTypeAndObjectId,
  KogitoSpinner,
  DataTable
} from '@kogito-apps/common';
import {
  ICell,
  ITransform,
  IFormatterValueType
} from '@patternfly/react-table';

const UserTaskLoadingComponent = (
  <Bullseye>
    <KogitoSpinner spinnerText="Loading user tasks..." />
  </Bullseye>
);

const stateColumnTransformer: ITransform = (value: IFormatterValueType) => {
  if (!value) {
    return null;
  }
  const { title } = value;
  return {
    children: <Label>{title}</Label>
  };
};

const UserTaskDataTableContainer: React.FC<InjectedOuiaProps> = ({
  ouiaContext
}) => {
  const {
    loading,
    error,
    data,
    refetch,
    networkStatus
  } = useGetUserTasksByStatesQuery({
    variables: {
      state: ['Ready'] // FIXME: state should not be hard-coded.
    },
    fetchPolicy: 'network-only',
    notifyOnNetworkStatusChange: true
  });

  const columns: ICell[] = [
    {
      title: 'ProcessId',
      data: 'processId'
    },
    {
      title: 'Name',
      data: 'name'
    },
    {
      title: 'Priority',
      data: 'priority'
    },
    {
      title: 'ProcessInstanceId',
      data: 'processInstanceId'
    },
    {
      title: 'State',
      data: 'state',
      cellTransforms: [stateColumnTransformer]
    }
  ];

  useEffect(() => {
    return ouiaPageTypeAndObjectId(ouiaContext, 'user-tasks');
  });

  return (
    <React.Fragment>
      <UserTaskPageHeader />
      <PageSection>
        <Grid gutter="md">
          <GridItem span={12}>
            <Card className="kogito-task-console--user-task_table-OverFlow">
              <DataTable
                data={data ? data.UserTaskInstances : undefined}
                isLoading={loading}
                columns={columns}
                networkStatus={networkStatus}
                error={error}
                refetch={refetch}
                LoadingComponent={UserTaskLoadingComponent}
              />
            </Card>
          </GridItem>
        </Grid>
      </PageSection>
    </React.Fragment>
  );
};

export default withOuiaContext(UserTaskDataTableContainer);
