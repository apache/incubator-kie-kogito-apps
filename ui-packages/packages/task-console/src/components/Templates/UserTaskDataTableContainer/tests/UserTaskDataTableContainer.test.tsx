import React from 'react';
import UserTaskDataTableContainer from '../UserTaskDataTableContainer';
import { getWrapperAsync } from '@kogito-apps/common';
import { MemoryRouter as Router } from 'react-router-dom';
import { MockedProvider } from '@apollo/react-testing';

const MockedComponent = (): React.ReactElement => {
  return <></>;
};
// tslint:disable-next-line: no-var-requires
const mockGraphqlTypes = require('../../../../graphql/types');

jest.mock('../../../Molecules/UserTaskPageHeader/UserTaskPageHeader');
jest.mock('@kogito-apps/common', () => ({
  ...jest.requireActual('@kogito-apps/common'),
  DataTable: () => {
    return <MockedComponent />;
  }
}));
jest.unmock('../../../../graphql/types');

const mockUserTaskInstancesData = {
  UserTaskInstances: [
    {
      __typename: 'UserTaskInstance',
      actualOwner: null,
      adminGroups: [],
      adminUsers: [],
      completed: null,
      description: '',
      excludedUsers: [],
      id: '047ec38d-5d57-4330-8c8d-9bd67b53a529',
      inputs:
        '{"flight":{"flightNumber":"MX555","seat":null,"gate":null,"departure":"2019-12-09T23:00:00.000+01:00","arrival":"2019-12-14T23:00:00.000+01:00"},"TaskName":"ConfirmTravel","NodeName":"Confirm travel","Priority":"1","Skippable":"true","hotel":{"name":"Perfect hotel","address":{"street":"street","city":"New York","zipCode":"12345","country":"US"},"phone":"09876543","bookingNumber":"XX-012345","room":null}}',
      name: 'Confirm travel',
      outputs: '{"ActorId":""}',
      potentialGroups: [],
      potentialUsers: [],
      priority: '1',
      processId: 'travels',
      processInstanceId: '9ae407dd-cdfa-4722-8a49-0a6d2e14550d',
      referenceName: 'ConfirmTravel',
      rootProcessId: null,
      rootProcessInstanceId: null,
      started: '2020-02-19T10:59:34.185Z',
      state: 'Ready'
    },
    {
      __typename: 'UserTaskInstance',
      actualOwner: null,
      adminGroups: [],
      adminUsers: [],
      completed: null,
      description: null,
      excludedUsers: [],
      id: '45a73767-5da3-49bf-9c40-d533c3e77ef3',
      inputs:
        '{"Skippable":"true","trip":{"city":"Boston","country":"US","begin":"2020-02-19T23:00:00.000+01:00","end":"2020-02-26T23:00:00.000+01:00","visaRequired":true},"TaskName":"VisaApplication","NodeName":"Apply for visa","traveller":{"firstName":"Rachel","lastName":"White","email":"rwhite@gorle.com","nationality":"Polish","address":{"street":"Cabalone","city":"Zerf","zipCode":"765756","country":"Poland"}},"Priority":"1"}"',
      name: 'Apply for visa',
      outputs: '{}',
      potentialGroups: [],
      potentialUsers: [],
      priority: '1',
      processId: 'travels',
      processInstanceId: '9ae7ce3b-d49c-4f35-b843-8ac3d22fa427',
      referenceName: 'VisaApplication',
      rootProcessId: null,
      rootProcessInstanceId: null,
      started: '2020-02-19T11:11:56.282Z',
      state: null
    }
  ]
};

describe('UserTaskDataTableContainer component tests', () => {
  it('Should render UserTaskDataTableContainer correctly', async () => {
    mockGraphqlTypes.useGetUserTasksByStatesQuery = jest.fn().mockReturnValue({
      loading: false,
      error: undefined,
      refetch: jest.fn(),
      networkStatus: 1,
      data: mockUserTaskInstancesData
    });

    const wrapper = await getWrapperAsync(
      <Router>
        <MockedProvider>
          <UserTaskDataTableContainer />
        </MockedProvider>
      </Router>,
      'UserTaskDataTableContainer'
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render UserTaskDataTableContainer with no data', async () => {
    mockGraphqlTypes.useGetUserTasksByStatesQuery = jest.fn().mockReturnValue({
      loading: false,
      error: undefined,
      refetch: jest.fn(),
      networkStatus: 1
    });

    const wrapper = await getWrapperAsync(
      <Router>
        <MockedProvider>
          <UserTaskDataTableContainer />
        </MockedProvider>
      </Router>,
      'UserTaskDataTableContainer'
    );
    expect(wrapper).toMatchSnapshot();
    expect(mockGraphqlTypes.useGetUserTasksByStatesQuery).toBeCalled();
  });
});
