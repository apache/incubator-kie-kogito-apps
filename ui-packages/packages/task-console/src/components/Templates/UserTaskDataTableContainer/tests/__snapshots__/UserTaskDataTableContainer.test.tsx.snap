// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTaskDataTableContainer component tests Should render UserTaskDataTableContainer correctly 1`] = `
<UserTaskDataTableContainer
  ouiaContext={
    Object {
      "isOuia": false,
      "ouiaId": null,
    }
  }
>
  <MockUserTaskPageHeader />
  <PageSection>
    <section
      className="pf-c-page__main-section"
    >
      <Grid
        gutter="md"
      >
        <div
          className="pf-l-grid pf-m-gutter"
        >
          <GridItem
            span={12}
          >
            <div
              className="pf-l-grid__item pf-m-12-col"
            >
              <Card
                className="kogito-task-console--user-task_table-OverFlow"
              >
                <article
                  className="pf-c-card kogito-task-console--user-task_table-OverFlow"
                >
                  <DataTable
                    LoadingComponent={
                      <Bullseye>
                        <KogitoSpinner
                          spinnerText="Loading user tasks..."
                        />
                      </Bullseye>
                    }
                    columns={
                      Array [
                        Object {
                          "data": "processId",
                          "title": "ProcessId",
                        },
                        Object {
                          "data": "name",
                          "title": "Name",
                        },
                        Object {
                          "data": "priority",
                          "title": "Priority",
                        },
                        Object {
                          "data": "processInstanceId",
                          "title": "ProcessInstanceId",
                        },
                        Object {
                          "cellTransforms": Array [
                            [Function],
                          ],
                          "data": "state",
                          "title": "State",
                        },
                      ]
                    }
                    data={
                      Array [
                        Object {
                          "__typename": "UserTaskInstance",
                          "actualOwner": null,
                          "adminGroups": Array [],
                          "adminUsers": Array [],
                          "completed": null,
                          "description": "",
                          "excludedUsers": Array [],
                          "id": "047ec38d-5d57-4330-8c8d-9bd67b53a529",
                          "inputs": "{\\"flight\\":{\\"flightNumber\\":\\"MX555\\",\\"seat\\":null,\\"gate\\":null,\\"departure\\":\\"2019-12-09T23:00:00.000+01:00\\",\\"arrival\\":\\"2019-12-14T23:00:00.000+01:00\\"},\\"TaskName\\":\\"ConfirmTravel\\",\\"NodeName\\":\\"Confirm travel\\",\\"Priority\\":\\"1\\",\\"Skippable\\":\\"true\\",\\"hotel\\":{\\"name\\":\\"Perfect hotel\\",\\"address\\":{\\"street\\":\\"street\\",\\"city\\":\\"New York\\",\\"zipCode\\":\\"12345\\",\\"country\\":\\"US\\"},\\"phone\\":\\"09876543\\",\\"bookingNumber\\":\\"XX-012345\\",\\"room\\":null}}",
                          "name": "Confirm travel",
                          "outputs": "{\\"ActorId\\":\\"\\"}",
                          "potentialGroups": Array [],
                          "potentialUsers": Array [],
                          "priority": "1",
                          "processId": "travels",
                          "processInstanceId": "9ae407dd-cdfa-4722-8a49-0a6d2e14550d",
                          "referenceName": "ConfirmTravel",
                          "rootProcessId": null,
                          "rootProcessInstanceId": null,
                          "started": "2020-02-19T10:59:34.185Z",
                          "state": "Ready",
                        },
                        Object {
                          "__typename": "UserTaskInstance",
                          "actualOwner": null,
                          "adminGroups": Array [],
                          "adminUsers": Array [],
                          "completed": null,
                          "description": null,
                          "excludedUsers": Array [],
                          "id": "45a73767-5da3-49bf-9c40-d533c3e77ef3",
                          "inputs": "{\\"Skippable\\":\\"true\\",\\"trip\\":{\\"city\\":\\"Boston\\",\\"country\\":\\"US\\",\\"begin\\":\\"2020-02-19T23:00:00.000+01:00\\",\\"end\\":\\"2020-02-26T23:00:00.000+01:00\\",\\"visaRequired\\":true},\\"TaskName\\":\\"VisaApplication\\",\\"NodeName\\":\\"Apply for visa\\",\\"traveller\\":{\\"firstName\\":\\"Rachel\\",\\"lastName\\":\\"White\\",\\"email\\":\\"rwhite@gorle.com\\",\\"nationality\\":\\"Polish\\",\\"address\\":{\\"street\\":\\"Cabalone\\",\\"city\\":\\"Zerf\\",\\"zipCode\\":\\"765756\\",\\"country\\":\\"Poland\\"}},\\"Priority\\":\\"1\\"}\\"",
                          "name": "Apply for visa",
                          "outputs": "{}",
                          "potentialGroups": Array [],
                          "potentialUsers": Array [],
                          "priority": "1",
                          "processId": "travels",
                          "processInstanceId": "9ae7ce3b-d49c-4f35-b843-8ac3d22fa427",
                          "referenceName": "VisaApplication",
                          "rootProcessId": null,
                          "rootProcessInstanceId": null,
                          "started": "2020-02-19T11:11:56.282Z",
                          "state": null,
                        },
                      ]
                    }
                    isLoading={false}
                    networkStatus={1}
                    refetch={[MockFunction]}
                  >
                    <MockedComponent />
                  </DataTable>
                </article>
              </Card>
            </div>
          </GridItem>
        </div>
      </Grid>
    </section>
  </PageSection>
</UserTaskDataTableContainer>
`;

exports[`UserTaskDataTableContainer component tests Should render UserTaskDataTableContainer with no data 1`] = `
<UserTaskDataTableContainer
  ouiaContext={
    Object {
      "isOuia": false,
      "ouiaId": null,
    }
  }
>
  <MockUserTaskPageHeader />
  <PageSection>
    <section
      className="pf-c-page__main-section"
    >
      <Grid
        gutter="md"
      >
        <div
          className="pf-l-grid pf-m-gutter"
        >
          <GridItem
            span={12}
          >
            <div
              className="pf-l-grid__item pf-m-12-col"
            >
              <Card
                className="kogito-task-console--user-task_table-OverFlow"
              >
                <article
                  className="pf-c-card kogito-task-console--user-task_table-OverFlow"
                >
                  <DataTable
                    LoadingComponent={
                      <Bullseye>
                        <KogitoSpinner
                          spinnerText="Loading user tasks..."
                        />
                      </Bullseye>
                    }
                    columns={
                      Array [
                        Object {
                          "data": "processId",
                          "title": "ProcessId",
                        },
                        Object {
                          "data": "name",
                          "title": "Name",
                        },
                        Object {
                          "data": "priority",
                          "title": "Priority",
                        },
                        Object {
                          "data": "processInstanceId",
                          "title": "ProcessInstanceId",
                        },
                        Object {
                          "cellTransforms": Array [
                            [Function],
                          ],
                          "data": "state",
                          "title": "State",
                        },
                      ]
                    }
                    isLoading={false}
                    networkStatus={1}
                    refetch={[MockFunction]}
                  >
                    <MockedComponent />
                  </DataTable>
                </article>
              </Card>
            </div>
          </GridItem>
        </div>
      </Grid>
    </section>
  </PageSection>
</UserTaskDataTableContainer>
`;
